src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1232,10 +1226,11 @@ void grpc_chttp2_complete_closure_step(grpc_chttp2_transport* t,
         write_state_name(t->write_state), whence.file(), whence.line());
   }
 
-  if (s->context != nullptr) {
-    MaybeRecordTransportAnnotation(
-        s, absl::StrFormat("on_complete: s=%p %p desc=%s err=%s", s, closure,
-                           desc, grpc_core::StatusToString(error).c_str()));
+  if (auto* tracer = CallTracerIfEnabled(s)) {
+    tracer->RecordAnnotation(
+        absl::StrCat("on_complete: s=0x", absl::Hex(s, absl::kZeroPad16), " 0x",
+                     absl::Hex(closure, absl::kZeroPad16), " desc=", desc,
```

Comment:
Given that the string is now only generated when tracing is enabled, that should sufficiently mitigate the performance impact, yes?

If so, I'd recommend sticking with `StrFormat("%p", s)` here for readability, and consistency with the ~1700 other places in the codebase where `%p` is used to print addresses.

If not, and this StrCat + `absl::Hex` trick is necessary for performance, we may want to make this change more broadly.

Reply:
Reverted. I left the changes to transport_op_string to move to appending instead of building up a vector and joining but switched some of them back to format string via StrAppendFormat.